### PR TITLE
Refactor `lookupNetwork` tests

### PR DESCRIPTION
**Pull request by Gudahtt**
## Description

The `lookupNetwork` tests have been refactored in preparation for reusing these test cases for other methods (in #1318). The two separate blocks for Infura v.s. custom networks have been consolidated into one block, and some tests have been modified to no longer test for the absence of events (instead asserting the expected end-state expected).

## Changes

None

## References

This relates to #1210

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation for new or updated code as appropriate (note: this will usually be JSDoc)
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/network-controller/tests/NetworkController.test.ts [hunk 1]
```diff
@@ -461,298 +461,33 @@ describe('NetworkController', () => {
...
-            );
-          });
-        });
-
-        describe('if an RPC error is encountered while retrieving the version of the current network', () => {
-          it('updates the network in state to "unavailable"', async () => {
+    for (const providerConfig of Object.values(NetworkType).map((networkType) =>
+      buildProviderConfig({ type: networkType }),
+    )) {
+      describe('if the network ID and network details requests resolve successfully', () => {
```
[the remark this answers is not in the document]
Comment (Gudahtt) [reply packages/network-controller/tests/NetworkController.test.ts:467]: Good catch; this loop was just added as a way to break the tests up, the `describe` block per-network is restored in a later PR. I hadn't realized it was removed here. I'll add it now

[the remark this answers is not in the document]
Comment (Gudahtt) [reply packages/network-controller/tests/NetworkController.test.ts]: Done in 92f0c69a

[the remark this answers is not in the document]
Comment (Gudahtt) [reply packages/network-controller/tests/NetworkController.test.ts]: Definitely agreed that this is useful; it's quite irritating when a test fails but it's not clear under what conditions.

This comment reminded me that I had this problem when looking at the network client tests recently! I fixed it locally but forgot to put it up as a PR, so here it is: https://github.com/MetaMask/core/pull/1321

**Review by mcmire — APPROVED** [review]
Looks good!

**Pull request merged**